ui/app/appointments/services/calendarViewPopUp.js

```diff
@@ -47,6 +48,25 @@ angular.module('bahmni.appointments')
                     closeConfirmBox();
                 };
 
+                var isCurrentUserHavePrivilege = function (privilege) {
+                    return !_.isUndefined(_.find($rootScope.currentUser.privileges, function (userPrivilege) {
+                        return userPrivilege.name === privilege;
+                    }));
+                };
+
+                popUpScope.isUserAllowedToPerform = function () {
+                    if (isCurrentUserHavePrivilege(popUpScope.manageAppointmentPrivilege)) {
+                        return true;
+                    }
+                    else if (isCurrentUserHavePrivilege(popUpScope.ownAppointmentPrivilege)) {
+                        return _.isEmpty(scope.appointments[0].providers) ||
+                            !_.isUndefined(_.find(scope.appointments[0].providers, function (provider) {
```

Comment:
@sai789,  can you please pull this block into a different method so that its readable